File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    
    _lost=[i for i in lost if i not in reserve]
    _reserve=[k for k in reserve if k not in lost]
    
    answer = n - len(_lost)
    _lost.sort()
    _reserve.sort()
    
    for i in _lost:
        if i-1 in _reserve:
            answer+=1
            _reserve.remove(i-1)
        elif i+1 in _reserve:
            answer+=1
            _reserve.remove(i+1)
    
    
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;
import java.io.*;
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        Arrays.sort(lost);
        Arrays.sort(reserve);
        
        int answer = n-(lost.length); //도단당해서 남은 학생수
        
        //도둑 맞으면서 체육복 가져왔으면 제외
        for(int i=0; i<lost.length; i++) {
            for(int j=0; j<reserve.length; j++) {
                if(lost[i] == reserve[j]) {
                    answer++;
                    lost[i] = -1;
                    reserve[j] =-1;
                    break;
                }
            }
        }
        
        

        for(int i=0; i<lost.length; i++) {
            for(int j=0; j<reserve.length; j++) {
                if( lost[i] -1 == reserve[j] || lost[i] +1 == reserve[j]) {
                    answer++;
                    reserve[j] =-1;
                    break;
                }
            }
        }
        
        return answer;
    }
}